project/npda/views/npda_users.py

```diff
@@ -722,68 +722,53 @@ def post(self, *args, **kwargs):
         # In local development, override the token workflow, just sign in
         # the user without 2FA token
         if settings.DEBUG:
-
             user = authenticate(
                 self.request,
                 username=self.request.POST.get("auth-username"),
                 password=self.request.POST.get("auth-password"),
             )
             if user is not None:
                 login(self.request, user)
-                # successful login, get PDU and organisation details from user and store in session
-
-                # Override normal auth flow behaviour, redirect straight to home page
-                audit_period = AuditPeriod.objects.get_default_audit_period()
 
-                return redirect(get_user_home_page(audit_period.slug, user))
+                # Override normal auth flow behaviour, redirect straight away
+                response = HttpResponseRedirect(self.get_success_url())
+                return self._done(response, user)
 
         # Otherwise, continue with usual workflow
         response = super().post(*args, **kwargs)
         return self.delete_cookies_from_response(response)
 
-    # Override successful login redirect to org summary page
-    def done(self, form_list, **kwargs):
-        # this will not be called if debug=True
-        response = super().done(form_list)
-        response_url = getattr(response, "url")
-
-        # redirect to home page
-        login_redirect_url = reverse(settings.LOGIN_REDIRECT_URL)
-
-        # Successful 2FA and login
-        if response_url == login_redirect_url:
```

Comment:
This is why you'd bypass the password expiry check if you logged in following a direct link. I don't really know why this was here, it appears to have been in the original 2fa bypass PR from E12 (https://github.com/rcpch/rcpch-audit-engine/pull/574)